subprojects/docs/src/docs/release/notes.md

```diff
@@ -121,6 +121,10 @@ task myTask {
 }
 ```
 
+### Deprecation of `TaskInternal.execute()`
+
+In this release we deprecate calling `TaskInternal.execute()`. Calling `task.execute()` should never be necessary. Tasks should be executed by requesting them from the command line or via the Tooling API. 
```

Comment:
I think this explanation will be unsatisfactory to users who do this. It's not that they don't know that tasks are executed via the command line. It's usually that they want to reuse the logic that some task is offering in another task by delegating to the first one. Or to do some custom parallelization.

We should explain how to replace some common patterns, e.g. using task dependencies, refactoring your tasks into a task class and a reusable piece of logic that others can use (think `Copy` and `project.copy`).

Reply:
I added a better message to the release notes. Could you have a look? Do you have a good idea what to say in the deprecation message?

Reply:
That looks great. The deprecation message is hard, ideally we'd have some documentation to link to. Maybe a short paragraph in the user guide?

Reply:
I added a section to the userguide and link from the deprecation message to it. WDYT?